text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+location, and GS1 prefixes, there are 60 hex characters remaining in the address.
+The 13 digits of the GLN can be left padded with 45-hex-character zeroes and 
+right padded with 2-hex-character zeroes to accommodate potential future storage
+associated with the GS1 Location representation, for example:
+
+    �621dee� + �04� + �01� + �000000000000000000000000000000000000000000000� +
+    13-character �numeric string� location_id + �00� // location_id == GLN
+
+A full GS1 Location address (for example purposes) would therefore be:
+
+    �621dee0401000000000000000000000000000000000000000000000123456789012800�
+
+
+## Transaction Payload and Execution
+[transaction payload and execution]: "transactionpayloadandexecution"
+
+### Location PayLoad Transaction
+
+LocationPayload contains an action enum and the associated action payload. This 
+allows for the action payload to be dispatched to the appropriate logic.
+
+Only the defined actions are available and only one action payload should be 
+defined in the LocationPayload.
+
+    message LocationPayload {        
+
+     enum Actions {
+        UNSET_ACTION = 0;
+        LOCATION_CREATE = 1;
+        LOCATION_UPDATE = 2;
+        LOCATION_DELETE = 3;
+    
+    }
+    
+    Action action = 1;
+    
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    LocationCreateAction location_create = 3;
+    LocationUpdateAction location_update = 4;
+    LocationDeleteAction location_delete = 5;
+    }
+
+### LocationCreateAction
+
+LocationCreateAction adds a new location to state. The transaction should be 
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the create transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+Validation requirements:
+- If a location with location_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must 
+belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_location for the organization, 
+otherwise the transaction is invalid.
+- If the location_namespace is GS1, the organization must contain a GS1 Company 
+Prefix in its metadata (gs1_company_prefixes), and the prefix must match the 
+company prefix in the location_id, which is a GLN if GS1, otherwise the 
+transaction is invalid.
+- The properties must be valid for the location_namespace. For example, if the
+location is GS1 location, its properties must only contain properties that are
+included in the GS1 Schema. If it includes a property not in the GS1 Schema the
+transaction is invalid.
+
+If all requirements are met, the transaction will be accepted, the batch will be
+written to a block, and the location will be created in state.
```

Comment:
Remove ", the batch will be written to a block" as this is DLT-specific (Splinter's Scabbard DLT does not have blocks, for example).